plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -129,6 +145,33 @@ public Page getNextPage()
         }
     }
 
+    @Override
+    public void deleteRows(Block rowIds)
+    {
+        if (positionDeleteSink == null) {
+            positionDeleteSink = positionDeleteSinkSupplier.get();
+        }
+        positionDeleteSink.appendPage(new Page(rowIds));
+    }
+
+    @Override
+    public CompletableFuture<Collection<Slice>> finish()
+    {
+        if (positionDeleteSink != null) {
+            return positionDeleteSink.finish();
+        }
+        return CompletableFuture.completedFuture(ImmutableList.of());
+    }
+
+    @Override
+    public void abort()
+    {
+        if (positionDeleteSink != null) {
+            positionDeleteSink.abort();
+        }
+        close();
+    }
+
     @Override
     public void close()
```

Comment:
- close `positionDeleteSink` 
- update `getMemoryUsage` too

Reply:
> close positionDeleteSink

Is close called on success or on failure?

Reply:
per ConnectorPageSource.close doc -- always

Reply:
Changed `abort` and `finish` to not call `close`. Looks like if the delete sink is present one of those two should always be called, and then `close` is called separately